# Rosalind
In this repository you'll find my solutions for bioinformatics problems on [ROSALIND](http://rosalind.info "ROSALIND"). I'll try to solve them in both Python and Java, but the Java solutions may take some time.

## Python Village
Here is an overview of all problems in the [Python Village](https://rosalind.info/problems/list-view/?location=python-village "Python Village") with links to the solutions in Python and Java.

| Problem ID | Name | Python Solution | Java Solution |
| :-----------: | :----: | :-------------: | :-------------: |
| INI1 | Installing Python | [Installing_Python.py](https://github.com/Kazouha/Rosalind/blob/63519e546881766bbe6e5c11440637db40b36909/Python_Village/Python/Installing_Python.py) | -- |
| INI2 | Variables and Some Arithmetic  | [Variables_and_some_Arithmetic.py](https://github.com/Kazouha/Rosalind/blob/9079418e46e625e5b62b7dd56929adffbc81e9d4/Python_Village/Python/Variables_and_some_Arithmetic.py) | VariablesAndArithmetic.java |
| INI3 | Strings and Lists | [Strings_and_Lists.py](https://github.com/Kazouha/Rosalind/blob/47ac17b342b95a52a91001fd5ff782763fc3b4a1/Python_Village/Python/Strings_and_Lists.py) | StringsAndLists.java |
| INI4 | Conditions and Loops | [Conditions_and_Loops.py](https://github.com/Kazouha/Rosalind/blob/440ffb60840b8d6a9d3153de4ff3df093bc04a9d/Python_Village/Python/Conditions_and_Loops.py) | ConditionsAndLoops.java|
| INI5 | Working with Files | [Working_with_Files.py](https://github.com/Kazouha/Rosalind/blob/440ffb60840b8d6a9d3153de4ff3df093bc04a9d/Python_Village/Python/Working_with_Files.py) | WorkingWithFiles.java |
| INI6 | Dictionaries | [Dictionaries.py](https://github.com/Kazouha/Rosalind/blob/ec83fc663ca43e85e110f38789317136b4f4f9ff/Python_Village/Python/Dictionaries.py) | Dictionaries.java

&nbsp;

## Bioinformatics Stronghold
Here is an overview of all problems in the [Bioinformatics Stronghold](https://rosalind.info/problems/list-view/) with links to the solutions in Python and Java.

| Problem ID | Name | Python Solution | Java Solution |
| :-----------: | :----: | :-------------: | :-------------: |
| DNA | Counting DNA Nucleotides | [Counting_Nucleotides.py](https://github.com/Kazouha/Rosalind/blob/2da441f423e634b844c83403146bbb8381a318f7/Bioinformatics_Stronghold/Python/Counting_Nucleotides.py) | CountingNucleotides.java |
| RNA | Transcribing DNA into RNA | [DNA_to_RNA.py](https://github.com/Kazouha/Rosalind/blob/2da441f423e634b844c83403146bbb8381a318f7/Bioinformatics_Stronghold/Python/DNA_to_RNA.py) | DNAToRNA.java |
| REVC | Complementing a Strand of DNA | Complement.py | Complemet.java |
| FIB | Rabbits and Recurrence Relations | Fib_Rabbits.py | FibRabbits.java |
| GC | Computing GC Content | GC_Content.py | GCContent.java |
| HAMM | Counting Point Mutations | Mutation_Count.py | MutationCount.java |
| IPRB | Mendel's First Law | Mendel.py | Mendel.java |
| PROT | Translating RNA into Protein | RNA_to_Protein.py | RNAToProtein.java |
| SUBS | Finding a Motif in DNA |  |  |  	  	 
| CONS | Consensus and Profile |  |  | 	  	 
| FIBD | Mortal Fibonacci Rabbits |  |  |	  	  	 
| GRPH | Overlap Graphs |  |  | 	 
| IEV | Calculating Expected Offspring |  |  |	  	  	 
| LCSM | Finding a Shared Motif |  |  |  	  	 
| LIA | Independent Alleles |  |  | 	  	  	 
| MPRT | Finding a Protein Motif |  |  |  	 
| MRNA | Inferring mRNA from Protein |  |  |	  	  	 
| ORF | Open Reading Frames |  |  |	  	  	 
| PERM | Enumerating Gene Orders |  |  |
| PRTM | Calculating Protein Mass |  |  |  	  	 
| REVP | Locating Restriction Sites |  |  |
| SPLC | RNA Splicing |  |  | 	  	  	 
| LEXF | Enumerating k-mers Lexicographically |  |  | 	 
| LGIS | Longest Increasing Subsequence |  |  |	  	  	 
| LONG | Genome Assembly as Shortest Superstring |  |  |  	 
| PMCH | Perfect Matchings and RNA Secondary Structures |  |  |	  	 
| PPER | Partial Permutations |  |  | 		  	  	 
| PROB | Introduction to Random Strings |  |  |		  	  	 
| SIGN | Enumerating Oriented Gene Orderings |  |  |		  	  	 
| SSEQ | Finding a Spliced Motif |  |  | 		  	  	 
| TRAN | Transitions and Transversions |  |  |	  	  	 
| TREE | Completing a Tree |  |  |		  	  	 
| CAT | Catalan Numbers and RNA Secondary Structures |  |  |		  	  	 
| CORR | Error Correction in Reads |  |  |	  	  	 
| INOD | Counting Phylogenetic Ancestors |  |  |		  	  	 
| KMER | k-Mer Composition |  |  |		  	  	 
| KMP |	Speeding Up Motif Finding |  |  |		  	  	 
| LCSQ | Finding a Shared Spliced Motif |  |  |		  	  	 
| LEXV | Ordering Strings of Varying Length Lexicographically |  |  |		  	  	 
| MMCH | Maximum Matchings and RNA Secondary Structures |  |  |		  	  	 
| PDST | Creating a Distance Matrix |  |  |		  	  	 
| REAR | Reversal Distance |  |  |		  	  	 
| RSTR | Matching Random Motifs |  |  | 		  	  	 
| SSET | Counting Subsets |  |  |	  	  	 
| ASPC | Introduction to Alternative Splicing |  |  |	  	  	 
| EDIT | Edit Distance |  |  |		  	  	 
| EVAL | Expected Number of Restriction Sites |  |  |		  	  	 
| MOTZ | Motzkin Numbers and RNA Secondary Structures |  |  |	  	  	 
| NWCK | Distances in Trees |  |  |	  	  	 
| SCSP | Interleaving Two Motifs |  |  |	  	  	 
| SETO | Introduction to Set Operations |  |  | 		  	  	 
| SORT | Sorting by Reversals |  |  | 	  	 
| SPEC | Inferring Protein from Spectrum |  |  | 		  	  	 
| TRIE | Introduction to Pattern Matching |  |  | 		  	  	 
| CONV | Comparing Spectra with the Spectral Convolution |  |  |		  	  	 
| CTBL | Creating a Character Table |  |  |		  	  	 
| DBRU | Constructing a De Bruijn Graph |  |  | 		  	  	 
| EDTA | Edit Distance Alignment |  |  | 		  	  	 
| FULL | Inferring Peptide from Full Spectrum |  |  |		  	  	 
| INDC | Independent Segregation of Chromosomes |  |  |		  	  	 
| ITWV | Finding Disjoint Motifs in a Gene |  |  |		  	  	 
| LREP | Finding the Longest Multiple Repeat |  |  | 		  	  	 
| NKEW | Newick Format with Edge Weights |  |  | 		  	  	 
| RNAS | Wobble Bonding and RNA Secondary Structures |  |  |		  	  	 
| AFRQ | Counting Disease Carriers |  |  |		  	  	 
| CSTR | Creating a Character Table from Genetic Strings |  |  | 		  	  	 
| CTEA | Counting Optimal Alignments |  |  |		  	  	 
| CUNR | Counting Unrooted Binary Trees |  |  |		  	  	 
| GLOB | Global Alignment with Scoring Matrix |  |  |		  	  	 
| PCOV | Genome Assembly with Perfect Coverage |  |  |	  	  	 
| PRSM | Matching a Spectrum to a Protein |  |  |		  	  	 
| QRT |	Quartets |  |  |	  	  	 
| SGRA | Using the Spectrum Graph to Infer Peptides |  |  | 		  	  	 
| SUFF | Encoding Suffix Trees |  |  |	  	  	 
| CHBP | Character-Based Phylogeny |  |  |		  	  	 
| CNTQ | Counting Quartets |  |  |		  	  	 
| EUBT | Enumerating Unrooted Binary Trees |  |  |		  	  	 
| GASM | Genome Assembly Using Reads |  |  |		  	  	 
| GCON | Global Alignment with Constant Gap Penalty |  |  |		  	  	 
| LING | Linguistic Complexity of a Genome |  |  |		  	  	 
| LOCA | Local Alignment with Scoring Matrix |  |  |		  	  	 
| MEND | Inferring Genotype from a Pedigree |  |  |	  	  	 
| MGAP | Maximizing the Gap Symbols of an Optimal Alignment |  |  |		  	  	 
| MREP | Identifying Maximal Repeats |  |  |	  	  	 
| MULT | Multiple Alignment |  |  |		  	  	 
| PDPL | Creating a Restriction Map |  |  |		  	  	 
| ROOT | Counting Rooted Binary Trees |  |  |		  	  	 
| SEXL | Sex-Linked Inheritance |  |  |		  	  	 
| SPTD | Phylogeny Comparison with Split Distance |  |  |		  	  	 
| WFMD | The Wright-Fisher Model of Genetic Drift |  |  |		  	  	 
| ALPH | Alignment-Based Phylogeny |  |  |		  	  	 
| ASMQ | Assessing Assembly Quality with N50 and N75 |  |  |		  	  	 
| CSET | Fixing an Inconsistent Character Set |  |  |		  	  	 
| EBIN | Wright-Fisher's Expected Behavior |  |  |		  	  	 
| FOUN | The Founder Effect and Genetic Drift |  |  |		  	  	 
| GAFF | Global Alignment with Scoring Matrix and Affine Gap Penalty |  |  | 		  	  	 
| GREP | Genome Assembly with Perfect Coverage and Repeats |  |  |		  	  	 
| OAP |	Overlap Alignment |  |  |		  	  	 
| QRTD | Quartet Distance |  |  |		  	  	 
| SIMS | Finding a Motif with Modifications |  |  |		  	  	 
| SMGB | Semiglobal Alignment |  |  |		  	  	 
| KSIM | Finding All Similar Motifs |  |  | 		  	  	 
| LAFF | Local Alignment with Affine Gap Penalty |  |  |		  	  	 
| OSYM | Isolating Symbols in Alignments |  |  |		  	  	 
| RSUB | Identifying Reversing Substitutions |  |  | 

